bip-schnorr.mediawiki

```diff
@@ -147,15 +150,19 @@ All provided signatures are valid with overwhelming probability if and only if t
 ==== Signing ====
 
 Input:
-* The secret key ''d'': an integer in the range ''1..n-1''.
+* The secret key ''d''': an integer in the range ''1..n-1''.
 * The message ''m'': a 32-byte array
+* The flag ''sign_even'' indicating whether the signature will be valid for the public key with even Y coordinate: a single byte.
 
 To sign ''m'' for public key ''dG'':
+* Let ''P' = bytes(dG)''
+* Let ''d = d''' if ''sign_even = 0'' or ''P'[0:1] == 0x02'', otherwise let ''d = -d'''.
```

Comment:
I think it's easier to follow if you use a "single assignment" style, where you first compute d, P, and then derive d' and P' from that (which are either equal to d, P, or their negations).

Reply:
This was intended to be single assignment style but I missed spaces. Fixing. But it's the other way round though to be consistent with `k`: The algorithm takes `d'`, computes `P'` and then derives `d` and `P`.

Reply:
fixed